src/tbp/monty/frameworks/config_utils/config_args.py

```diff
@@ -1153,6 +1154,197 @@ class FiveLMMontySOTAConfig(FiveLMMontyConfig):
+            dictionary of connectivity matrices given a number of learning modules.
+            In particular, it must return a dictionary with keys "sm_to_agent_dict",
+            "sm_to_lm_matrix", "lm_to_lm_matrix", and "lm_to_lm_vote_matrix". Defaults
+            to `make_multi_lm_flat_dense_connectivity`.
+
+    Returns:
+        `MontyConfig`: complete monty config for multi-LM experiment.
+    """
+    # Make learning module configs.
+    if learning_module_args is None:
+        learning_module_args = {}
+    learning_module_configs = {}
+    for i in range(n_lms):
+        lm_args_i = copy.deepcopy(learning_module_args)
+        # Rename specs keyed with "patch" to "patch_{i}".
+        for name in ["graph_delta_thresholds", "tolerances", "feature_weights"]:
+            if name in lm_args_i:
+                spec = lm_args_i[name]
+                if "patch" in spec:
+                    spec[f"patch_{i}"] = spec.pop("patch")
+        learning_module_configs[f"learning_module_{i}"] = {
+            "learning_module_class": learning_module_class,
+            "learning_module_args": lm_args_i,
+        }
+
+    # Make sensor module configs.
+    if sensor_module_args is None:
+        sensor_module_args = {}
+    sensor_module_configs = {}
+    for i in range(n_lms):
+        sm_args_i = copy.deepcopy(sensor_module_args)
+        sm_args_i["sensor_module_id"] = f"patch_{i}"
+        sensor_module_configs[f"sensor_module_{i}"] = {
+            "sensor_module_class": sensor_module_class,
+            "sensor_module_args": sm_args_i,
+        }
+    sensor_module_configs["view_finder"] = {
+        "sensor_module_class": DetailedLoggingSM,
+        "sensor_module_args": {
+            "sensor_module_id": "view_finder",
+            "save_raw_obs": sensor_module_args.get("save_raw_obs", False),
+        },
+    }
+
+    # Make motor system config.
+    if motor_system_args is None:
+        motor_system_args = {}
+    else:
+        motor_system_args = copy.deepcopy(motor_system_args)
+    motor_system_config = {
+        "motor_system_class": motor_system_class,
+        "motor_system_args": motor_system_args,
+    }
+
+    connectivity = connectivity_func(n_lms)
+
+    if monty_args is None:
+        monty_args = MontyArgs()
+    elif isinstance(monty_args, MontyArgs):
+        monty_args = copy.deepcopy(monty_args)
+    else:
+        monty_args = MontyArgs(**monty_args)
+
+    monty_config = MontyConfig(
+        monty_class=monty_class,
+        learning_module_configs=learning_module_configs,
+        sensor_module_configs=sensor_module_configs,
+        motor_system_config=motor_system_config,
+        sm_to_agent_dict=connectivity["sm_to_agent_dict"],
+        sm_to_lm_matrix=connectivity["sm_to_lm_matrix"],
+        lm_to_lm_matrix=connectivity["lm_to_lm_matrix"],
+        lm_to_lm_vote_matrix=connectivity["lm_to_lm_vote_matrix"],
+        monty_args=monty_args,
+    )
+    return monty_config
+
+
+"""
+etc.
```

Comment:
What's this for?

Reply:
Just a habit. I tend to break files into sections separated by double-commented section descriptions and this was not-very-helpful one. Banished!